multicluster/cmd/multicluster-controller/gateway_webhook.go

```diff
@@ -47,13 +49,21 @@ func (v *gatewayValidator) Handle(ctx context.Context, req admission.Request) ad
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 
+	ui := req.UserInfo
+	_, saName, err := serviceaccount.SplitUsername(ui.Username)
+	if err != nil {
+		klog.ErrorS(err, "Error getting ServiceAccount name", "Gateway", req.Namespace+"/"+req.Name)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	if saName != mcControllerSAName {
+		return admission.Errored(http.StatusPreconditionFailed, fmt.Errorf("Gateway can only be created or updated by Antrea Multi-cluster Controller"))
+	}
 	// Check if there is any existing Gateway.
 	gatewayList := &mcv1alpha1.GatewayList{}
```

Comment:
If we check only MC Controller can create Gateways, even this check is not necessary? @luolanzone

Reply:
Yeah, I feel we can remove this.

Reply:
The "unique Gateway" check is removed, as well as the unit test.